Dockerfile/Dockerfile

```diff
@@ -0,0 +1,71 @@
+# Ubuntu Trusty Docker file for base TileDB Dependencies
+
+# This Dockerfile copies files from the parent directory, requiring a build
+# context from the parent directory. From `./..`, execute the following to
+# build:
+# docker build -f Dockerfile/Dockerfile -t tiledb:dev .
+#
+# Use the build arg 'enable' to configure optional TileDB components, e.g.:
+#   docker build -f Dockerfile/Dockerfile --build-arg enable=s3,hdfs -t tiledb:dev
+
+# Ubuntu Trusty
+FROM ubuntu:18.04
+
+# Optional components to enable (defaults to empty).
+ARG enable
+
+# Setup home environment
+RUN useradd tiledb
+RUN mkdir /home/tiledb && chown tiledb /home/tiledb
+ENV HOME /home/tiledb
+
+RUN apt-get update && apt-get install -y \
+    build-essential \
+    wget \
+    unzip \
+    git \
+    cmake \
+    python3 \
+    python3-dev \
+    python3-pip \
+    libssl-dev \
+    && apt-get clean \
+    && apt-get purge -y \
+    && rm -rf /var/lib/apt/lists*
+
+RUN cd /tmp \
+    && wget https://github.com/Kitware/CMake/releases/download/v3.17.2/cmake-3.17.2-Linux-x86_64.tar.gz \
+    && tar -xzf cmake-3.17.2-Linux-x86_64.tar.gz \
+    && cp -R cmake-3.17.2-Linux-x86_64/bin /usr/ \
+    && cp -R cmake-3.17.2-Linux-x86_64/doc /usr/ \
+    && cp -R cmake-3.17.2-Linux-x86_64/man /usr/ \
+    && cp -R cmake-3.17.2-Linux-x86_64/share /usr/
+
+# Copy TileDB source files from the build context to `/home/tiledb/TileDB``
+COPY CMakeLists.txt /home/tiledb/TileDB/CMakeLists.txt
+COPY bootstrap /home/tiledb/TileDB/bootstrap
+COPY cmake /home/tiledb/TileDB/cmake
+COPY examples /home/tiledb/TileDB/examples
+COPY external /home/tiledb/TileDB/external
+COPY scripts /home/tiledb/TileDB/scripts
+COPY test /home/tiledb/TileDB/test
+COPY tiledb /home/tiledb/TileDB/tiledb
```

Comment:
Consider including `tools` as well, and build it below?

Or since almost everything is added here, could use `ADD . /home/tiledb/TileDB` instead -- probably with a `.dockerignore` for excluding `.git`.

Reply:
Added tools to the copy and build.

The primary motivation for a whitelist is to keep the dockerfile as small as possible, and to avoid accidentally packaging something else in the future. For example, if we add a dedicated `benchmarks` folder or something.

The main thing is to ignore the build directory.

Reply:
One reason to use a single call is that each statement in a Dockerfile creates a new layer. It's not that big a deal for a small image, but just noting that selective inclusion can also be done with `.dockerignore`, eg:

```
*

!external/*
!scripts/*
!test/*
...
```

(via https://docs.docker.com/engine/reference/builder/#dockerignore-file)